community/io/src/main/java/org/neo4j/io/mem/MemoryAllocator.java

```diff
@@ -50,4 +50,9 @@ static MemoryAllocator createAllocator( String expectedMemory, MemoryAllocationT
      * @throws OutOfMemoryError if the requested memory could not be allocated.
      */
     long allocateAligned( long bytes, long alignment );
+
+    /**
+     * Close all allocated resources and free all allocated memory.
+     */
+    void close();
```

Comment:
I think we could specify that the resources will be freed by the GC as well, if not done explicitly, and also warn about use-after-free problems.